cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -164,6 +167,18 @@ func NewStaticAutoscaler(
 	clusterStateRegistry := clusterstate.NewClusterStateRegistry(autoscalingContext.CloudProvider, clusterStateConfig, autoscalingContext.LogRecorder, backoff, clusterstate.NewDefaultMaxNodeProvisionTimeProvider(autoscalingContext, processors.NodeGroupConfigProcessor))
 	processors.ScaleDownCandidatesNotifier.Register(clusterStateRegistry)
 
+	scaleDownCandidatesComparers := []scaledowncandidates.CandidatesComparer{}
```

Comment:
I have just moved this code from [here](https://github.com/kubernetes/autoscaler/pull/5729/files#diff-76f2867167fb87a2c61e82ea733b893d48724ca43e861b86a96f6dc582805ad4L418-L428) so that I can pass `clusterStateRegistry` only to `ScaleDownCandidatesDelayProcessor` [here](https://github.com/kubernetes/autoscaler/pull/5729/files#diff-0fa955fbc2ad636c52cbe796dda67c63f91ae5348d72c1d4c420f493a63cfb00R39). Another option is to change the interface which the processor implements so pass down an extra parameter.

Passing `clusterStateRegistry` to the processor might not be the best idea because the processor has access to the entire cluster state. Better ideas are welcome :pray:

Reply:
Maybe an intermediate interface which exposes only the required functions from `clusterStateRegistry` might be a better idea here.

Reply:
Static autoscaler shouldn't be responsible for selecting the right set of scale down processors (nor any other processors).